tests/unit/core/scan/test_scan_context.py

```diff
@@ -40,13 +49,71 @@ def test_scan_context_repository_url_parsed(self, tmp_path: Path):
         THEN the remote url is found and simplified
         """
         local_repo = Repository.create(tmp_path)
-        remote_url = "https://user:password@github.com:84/owner/repository.git"
-        expected_url = "github.com/owner/repository"
-        local_repo.git("remote", "add", "origin", remote_url)
+        local_repo.git("remote", "add", "origin", self.FULL_REPO_URL)
 
         context = ScanContext(
             scan_mode=ScanMode.PATH,
             command_path="ggshield secret scan path",
             target_path=local_repo.path,
         )
-        self._assert_repo_url_in_headers(context, expected_url)
+        self._assert_repo_url_in_headers(context, self.EXPECTED_HEADER)
+
+    def test_ci_repo_found(self, tmp_path: Path) -> None:
```

Comment:
This test and `test_ci_no_env()` reimplement pytest parametrized tests. I assume this is to avoid recreating the repo for each iteration, but this can be avoided using module-level fixtures.

Can you rewrite them using parametrized tests? The code could look something like this (untested):


```python
@pytest.fixture(scope="module")
def fake_url_repo(tmp_path_factory: pytest.TempPathFactory) -> Repository:
    repo = Repository.create(tmp_path_factory.mktemp("fake_url_repo"))
    repo.git("remote", "add", "origin", FAKE_URL)
    return repo


@pytest.mark.parametrize(
    "env",
    (
        {"BUILD_BUILDID": "1", "BUILD_REPOSITORY_URI": REPO_URL},
        {"DRONE": "1", "DRONE_REPO_LINK": REPO_URL},
        {
            "GITHUB_ACTIONS": "1",
            "GITHUB_SERVER_URL": DOMAIN,
            "GITHUB_REPOSITORY": SLUG,
        },
        {"GITLAB_CI": "1", "CI_REPOSITORY_URL": REPO_URL},
        {"CIRCLECI": "1", "CIRCLE_REPOSITORY_URL": REPO_URL},
        {"BITBUCKET_COMMIT": "1", "BITBUCKET_GIT_HTTP_ORIGIN": REPO_URL},The advan
    )
)
def test_ci_repo_found(env, fake_url_repo) -> None:
    with mock.patch.dict(os.environ, env, clear=True):
        context = ScanContext(
            scan_mode=ScanMode.CI,
            command_path="ggshield secret scan path",
            target_path=local_repo.path,
        )

    _assert_repo_url_in_headers(context, EXPECTED_HEADER)
```

The advantage of this is you get more precise error messages in case of test failures and you have the ability to re-run only selected parameters.

Reply:
The module scoped fixture is indeed exactly what I needed, thank you!